lldb/source/Core/PluginManager.cpp

```diff
@@ -965,6 +965,16 @@ Status PluginManager::SaveCore(const lldb::ProcessSP &process_sp,
     return error;
   }
 
+  if (!options.GetProcess())
+    options.SetProcess(process_sp);
```

Comment:
We should be doing this before calling SaveCore as what if this process doesn't match the process that is on the options. So have the code that calls this fill in the process ahead of time in the `lldb_private::SaveCoreOptions` instance. So always extract the process from the `lldb_private::SaveCoreOptions` instance and don't pass a process into this function. The code that calls this will need to ensure the process instances match.